pkg/errors/broadcasting_test.go

```diff
@@ -0,0 +1,49 @@
+package errors_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	broadcastError "github.com/bsv-blockchain/go-wallet-toolbox/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBroadcastingError_Is(t *testing.T) {
+	t.Run("should support errors.Is with underlying error", func(t *testing.T) {
+		// given:
+		originalErr := fmt.Errorf("connection timeout")
+		wrappedErr := fmt.Errorf("service failed: %w", originalErr)
+		broadcastErr := broadcastError.NewBroadcastingError(wrappedErr, broadcastError.CreateAction)
+		anotherBroadcastErr := &broadcastError.BroadcastingError{}
+
+		// when:
+		isOriginal := errors.Is(broadcastErr, originalErr)
+		isWrapped := errors.Is(broadcastErr, wrappedErr)
+		isType := errors.Is(broadcastErr, anotherBroadcastErr)
+
+		// then:
+		assert.True(t, isOriginal)
+		assert.True(t, isWrapped)
+		assert.True(t, isType)
```

Comment:
I would go with something like that:
```suggestion
		// expect:
		assert.True(t, errors.Is(broadcastErr, wrappedErr))
		
		// and:
		assert.True(t, errors.Is(broadcastErr, anotherBroadcastErr))
		
		// and:
		assert.False(t, errors.Is(broadcastErr, errors.New("any")))
```

Reply:
Maybe also add some message in case of failure to make it easier to distinguish which assertion failed. 
like:
```go
assert.True(t, errors.Is(broadcastErr, wrappedErr), "BroadcastingError is also an error from that cause it")
```

Reply:
added